documentation/src/main/asciidoc/public/reference/_static-metamodel-processor.adoc

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Red Hat Inc. and Hibernate Authors
+[[static-metamodel-processor]]
+= Annotation processor
+
+include::../components/_incubating-warning.adoc[]
+
+[[static-metamodel-processor-enabling]]
+== Enabling the annotation processor
+
+Hibernate Search provides a dedicated annotation processor to generate the static metamodel classes.
+This annotation processor is located in the `org.hibernate.search:hibernate-search-metamodel-processor`.
+
+The annotation processor has to be added to the build, e.g. for Maven:
+
+====
+[source,XML,subs="+attributes"]
+----
+<plugin>
+    <artifactId>maven-compiler-plugin</artifactId>
+    <executions>
+        <execution>
+            <id>default-compile</id>
+            <configuration>
+                <annotationProcessors>
+                    <annotationProcessor>org.hibernate.search.metamodel.processor.HibernateSearchMetamodelProcessor</annotationProcessor> <1>
+                </annotationProcessors>
+                <annotationProcessorPaths>
+                    <path>
+                        <groupId>org.hibernate.search</groupId>
+                        <artifactId>hibernate-search-metamodel-processor</artifactId> <2>
+                    </path>
+                    <path>
+                        <groupId>org.hibernate.search</groupId>
+                        <artifactId>hibernate-search-backend-lucene</artifactId> <3>
+                    </path>
+                </annotationProcessorPaths>
+            </configuration>
+        </execution>
+    </executions>
+</plugin>
+----
+
+<1> Provide the fully qualified class name of the annotation processor that generates the metamodel.
+<2> Add the `org.hibernate.search:hibernate-search-metamodel-processor` dependency as an annotation processor path for the Maven compiler plugin to be able to actually find the processor.
+<3> Add the backend dependency, in this example, the <<backend-lucene,Lucene backend>>, as an annotation processor path.
+It is important to include the same backend that the application is using to make sure that the generated metamodel classes reflect all the backend specifics.
+For example, backends might have different defaults, resulting in a different set of field traits depending on the backend.
+
+NOTE: The version of both annotation processor and backend dependencies are skipped in the definition of the annotation paths, as in case the Hibernate Search BOM is imported via dependency management, the compiler plugin will use the managed versions.
```

Comment:
```suggestion
NOTE: The version of both annotation processor and backend dependencies can be ommitted in the definition of the annotation paths, because they are defined in the Hibernate Search BOM, which we recommend you import via dependency management.
```